src/datasets/client.rs

```diff
@@ -132,8 +132,11 @@ impl Client {
         I: IntoIterator<Item = E>,
         E: Serialize,
     {
-        let events: Vec<E> = events.into_iter().collect();
-        let json_payload = serde_json::to_vec(&events)?;
+        let json_lines: Result<Vec<Vec<u8>>> = events
+            .into_iter()
+            .map(|event| serde_json::to_vec(&event).map_err(|e| Error::Serialize(e)))
```

Comment:
There is a comment from clippy here, I think this could be `map_err(Error::Serialize)`

Reply:
Nice one, fixed